pulp_smash/tests/rpm/api_v2/test_signatures_checked_for_syncs.py

```diff
@@ -0,0 +1,454 @@
+            (RPM_FEED_URL, 'rpm'),
+            (SRPM_FEED_URL, 'srpm'),
+            (RPM_UNSIGNED_FEED_URL, 'rpm'),
+            (DRPM_UNSIGNED_FEED_URL, 'drpm'),
+            (SRPM_UNSIGNED_FEED_URL, 'srpm'),
+        )
+        for feed, type_id in variants:
+            with self.subTest(type_id=type_id):
+                cu_counts = self.create_sync_repo({
+                    'allowed_keys': ['01234567'],
+                    'feed': feed,
+                    'require_signature': True,
+                })['content_unit_counts']
+                self.assertNotIn(type_id, cu_counts, cu_counts)
+
+
+class RequireAnyKeyTestCase(_BaseTestCase):
+    """Use an importer that requires signatures and has no key IDs.
+
+    The importer should have the following pseudocode configuration:
+
+    .. code-block:: json
+
+        {"require_signature": true, "allowed_keys": []}
+    """
+
+    def test_signed_rpm(self):
+        """Sync signed RPMs into the repository.
+
+        Assert packages are synced in.
+        """
+        cu_counts = self.create_sync_repo({
+            'allowed_keys': [],
+            'feed': RPM_FEED_URL,
+            'require_signature': True,
+        })['content_unit_counts']
+        self.assertEqual(cu_counts['rpm'], RPM_FEED_COUNT, cu_counts)
+
+    def test_signed_srpm(self):
+        """Sync signed SRPMs into the repository.
+
+        Assert packages are synced in.
+        """
+        cu_counts = self.create_sync_repo({
+            'allowed_keys': [],
+            'feed': SRPM_FEED_URL,
+            'require_signature': True,
+        })['content_unit_counts']
+        self.assertEqual(cu_counts['srpm'], SRPM_FEED_COUNT, cu_counts)
+
+    def test_unsigned_packages(self):
+        """Sync unsigned RPMs, DRPMs and SRPMS into repositories.
+
+        Assert no packages are synced in.
+        """
+        variants = (
+            (RPM_UNSIGNED_FEED_URL, 'rpm'),
+            (DRPM_UNSIGNED_FEED_URL, 'drpm'),
+            (SRPM_UNSIGNED_FEED_URL, 'srpm'),
+        )
+        for feed, type_id in variants:
+            with self.subTest(type_id=type_id):
+                cu_counts = self.create_sync_repo({
+                    'allowed_keys': [],
+                    'feed': feed,
+                    'require_signature': True,
+                })['content_unit_counts']
+                self.assertNotIn(type_id, cu_counts, cu_counts)
+
+
+class AllowInvalidKeyTestCase(_BaseTestCase):
+    """Use an importer that allows unsigned packages and has an invalid key ID.
+
+    The importer should have the following pseudocode configuration:
+
+    .. code-block:: json
+
+        {"require_signature": false, "allowed_keys": ["invalid key id"]}
+    """
+
+    def test_signed_packages(self):
+        """Import signed RPMs, DRPMs and SRPMs into repositories.
+
+        Assert no packages are synced.
+        """
+        variants = (
+            (RPM_FEED_URL, 'rpm'),
+            (SRPM_FEED_URL, 'srpm'),
```

Comment:
Ah. I need to fix the docstring. Yes, DRPMs should be a part of this test, and several others besides. However, we don't currently have signed DRPMs in pulp-fixtures. See: https://github.com/PulpQE/pulp-fixtures/issues/25